flash/core/data/utilities/collate.py

```diff
@@ -26,7 +26,8 @@ def _wrap_collate(collate: Callable, batch: List[Any]) -> Any:
     collated_batch = collate(batch)
 
     if metadata and isinstance(collated_batch, dict):
-        collated_batch[DataKeys.METADATA] = metadata
+        collated_batch["metadata"] = metadata
```

Comment:
The `DataKeys.METADATA` should still work here I think (the enum is meant to be interchangeable with a plain string)

Reply:
You are right. One question, @ethanwharris - once the `datamodule` is created, we don't expect to see the enums after this PR, right? The reason I ask this is - because I can see the Enums even after this change in the PR, for the `datamodule.train_dataloader().dataset`.

Reply:
Yeah, after the change you just made I would expect the dataset to not have the enums. Note that this may only be after the most recent change. Before that I think the default `train_dataloader` implementation doesn't use our collate function but not entirely sure.

Reply:
Thank you for the clarification! Looks like once the `datamodule` is created, the following:

```python
[x.keys() for x in datamodule.train_dataloader().dataset]
```

Gives: a list of: `dict_keys([<DataKeys.INPUT: 'input'>, <DataKeys.TARGET: 'target'>, <DataKeys.METADATA: 'metadata'>])`, which is not what we want/expected, right? Note that this is from this branch, and from the most recent commit.

The functions we made the changes in, only get called once we call the `fit` function from the `Trainer`, as I think then this `_load_sample` method is called from the `InputBase`.

I'll work on replacing enums with strings to ensure that `datamodule` doesn't contain Enums even before the `fit` function is called. I'll update this PR tomorrow! :)

Reply:
The `Input` is just a torch dataset. So the `datamodule.train_dataloader().dataset` should be an instance of `InputBase`. The `load_sample` is called in the `getitem` here:
https://github.com/Lightning-AI/lightning-flash/blob/0e2125926d10207ce7e1f905d4c392a465558651/flash/core/data/io/input.py#L277

So in theory if the enums are removed there then they shouldn't turn up in the dataset, but it's possible I'm missing something.

Reply:
Thanks, @ethanwharris. After investigation, `to_samples` method needed changes, but that only changes `input, target` keys. For `metadata`, every time `_call_load_sample` is called, a `<running_stage>_load_sample` method is called which eventually calls `load_sample` method (https://github.com/Lightning-AI/lightning-flash/blob/0e2125926d10207ce7e1f905d4c392a465558651/flash/core/data/io/input.py#L240).

Changes had to be made to the `load_sample` methods for each task, as they are overridden for each task. I'm thinking to add a test to make sure that this isn't missed out on any task. Will do this tomorrow, but hopefully the changes look good so far?